agent-core/src/main/java/com/newrelic/agent/android/payload/PayloadSender.java

```diff
@@ -107,14 +111,15 @@ protected void onFailedUpload(String errorMsg) {
     @SuppressWarnings("NewApi")
     public PayloadSender call() throws Exception {
         try {
-            timer.tic();
+            byte[] payloadBytes = getPayload().getBytes();
```

Comment:
get all the bytes might be a problem here

Reply:
Might be a problem in low memory conditions, we'll have to see. I will look for a more resilient solution.